src/gen/module.rs

```diff
@@ -105,9 +106,17 @@ impl Module {
     }
     /// Add submodule
     pub fn add_submodule(&mut self, module: Module) -> &mut Self {
-        self.sub_modules.push(module);
+        self.sub_modules.insert(module.name.clone(), module);
         self
     }
+    /// Get a mutable reference to a submodule of this module
+    pub fn get_submodule_mut(&mut self, name: impl ToString) -> Option<&mut Module> {
```

Comment:
Can we use the same bound as https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get `Borrow<T>` for these methods?

Reply:
I must be doing something silly.
```rust
pub fn get_submodule_mut<Q: ?Sized, K>(&mut self, name: &Q) -> Option<&mut Module>
        where
            K: Borrow<Q>,
            Q: Hash + Eq,
            String: Borrow<Q>
```

Results in 
```
assert!(module.get_submodule_mut("Bar2").is_none());
   |                    ^^^^^^^^^^^^^^^^^ cannot infer type for `K`

```

Reply:
The not mut version should look like this
```rust 
pub fn get_submodule<Q>(&self, name: &Q) -> Option<&String>
    where
        String: Borrow<Q>,
        Q: ?Sized + Hash + Eq,

Reply:
Sweet, thank you, sir!